# John Nicoludis

San Francisco, CA - jacknicoludis@gmail.com 

## Education 
**Harvard University**,   PhD, Chemistry and Chemical Biology,   2018  
**Swarthmore College**,   BA, Biochemistry,                      2012 

## Skills
### Languages  
Proficient: Python, UNIX/Linux shell scripting  
Familiar: MATLAB, R

### Data Analysis  
Dimensionality reduction  
Data visualization  
Hierarchical and K-means clustering  
Regression  
Classification

### Tools and Software  
Pandas, Jupyter, Scikit-learn, Biopython, ProDy,  
PyMOL, Phenix, Rosetta, VMD, Relion

### Computational Biology  
Structural biology  
Structure-based analysis, molecular dynamics  
Bioinformatics (alignment, coevolution)  
Computational protein design  
Analysis of microbiome NGS sequencing

## Experience

### Health Data Science Fellow (09/2019 - present)  
*Insight Fellows Program, San Francisco, CA*  
- Produced a pipeline for the analysis of microbiome compositional comparisons from raw 16S rRNA amplicon data  
- Trained a classifier to determine HIV-infection state microbiome data  
- Outperformed literature classifiers to distinguish HIV-negative and HIV-positive cohorts

### Postdoctoral Scholar (05/2018 – present)  
*Laboratory of Professor William DeGrado, University of California, San Francisco*  
- Developed Python pipelines to generate >100K conformational models of protein structures to query against database of protein structures and analyzed by hierarchical clustering  
- Learned Rosetta (computational protein design software) to design novel helical-assembling proteins  
- Characterized assembly of antimicrobial peptides with phospholipids with cryo-electron microscopy  
- Awarded NIH T32 Postdoctoral Fellowship in the Molecular and Cellular Basis of Cardiovascular Disease (>$60K)

### Teaching Fellow (08/2016 – 12/2016)  
*Biological Data Analysis, Professor Sean Eddy, Harvard University*  
- Led recitation sessions to 10-15 students on Python, RNA-seq data analysis, and statistical analysis  
- Generated synthetic data sets to be used in student problem sets  
- Developed lesson plans on dimensionality reduction and lectured to 60+ undergraduates

### Graduate Student Researcher (09/2012 – 05/2018)
*Laboratory of Professor Rachelle Gaudet, Harvard University*  
- Led team of 3 graduate students in different laboratories to publish a statistical model of protein-protein interaction specificity using molecular dynamics simulations, crystallography and sequence coevolution analysis  
- Authored 4 first author (PNAS, eLife, Structure, BBA Biomembranes) and 1 second author publication (eLife)  
- Mentored undergraduate student to write Python scripts to analyze conformational movements in proteins  
- Examined 100s of protein structures via dimensionality reduction to classify proteins into structural classes  
- Developed bioinformatic metric to determine residues contributing to protein-protein interaction specificity  
- Awarded National Defense Science and Engineering Graduate Fellow (>$100K)

### Undergraduate Student Researcher (03/2009 – 08/2012)
*Laboratory of Professor Liliya Yatsunyk, Swarthmore College*  
- Authored 2 first author (Nucleic Acids Research, JACS) and co-authored 4 other publications  
- Established DNA crystallography projects and mentored undergraduates on crystallography data analysis  
- Deconvolved biophysical signals (spectrophotometry, circular dichroism) from temperature series experiments by singular value decomposition to determine signals for short-lived intermediates
